llvm/lib/Target/ARM/ARMLegalizerInfo.cpp

```diff
@@ -467,6 +468,18 @@ bool ARMLegalizerInfo::legalizeCustom(LegalizerHelper &Helper, MachineInstr &MI,
     MIRBuilder.buildSetFPEnv(NewFPSCR);
     break;
   }
+  case G_RESET_FPMODE: {
+    // To get the default FP mode all control bits are cleared:
+    // FPSCR = FPSCR & (FPStatusBits | FPReservedBits)
+    LLT FPEnvTy = LLT::scalar(32);
+    auto FPEnv = MRI.createGenericVirtualRegister(FPEnvTy);
+    MIRBuilder.buildInstr(G_GET_FPENV).addDef({FPEnv});
+    auto NotModeBitMask = MIRBuilder.buildConstant(
```

Comment:
From scanning the legalizer I convinced myself that most opcodes that you build a legal. Now, that G_CONSTANT is custom, you can convince me that the constant is legal.

Reply:
The G_CONSTANT is illegal. When legalizer creates it in `legalizeInstrStep`, the object of type `LegalizerWorkListManager` adds this instruction to its worklist, and it will be subjected to legalization in next iterations of the legalization in `legalizeMachineFunction`.

Reply:
Yes, incremental legalization should be relied on. Legalize actions are not responsible for ensuring every output is legal